lib/Sema/CSSimplify.cpp

```diff
@@ -7316,6 +7367,99 @@ ConstraintSystem::simplifyConstructionConstraint(
   return SolutionKind::Solved;
 }
 
+ConstraintSystem::SolutionKind ConstraintSystem::simplifySubclassOfConstraint(
+                                 Type type,
+                                 Type classType,
+                                 ConstraintLocatorBuilder locator,
+                                 TypeMatchOptions flags) {
+  if (!classType->getClassOrBoundGenericClass())
```

Comment:
To be prudent I think we should `getFixedTypeRecursive` for `classType` as well.